libmamba/src/core/package_fetcher.cpp

```diff
@@ -433,16 +433,20 @@ namespace mamba
         std::ifstream index_file = open_ifstream(index_path);
         index_file >> index;
 
-        const nlohmann::json solvable_json = m_package_info;
-        index.insert(solvable_json.cbegin(), solvable_json.cend());
+        nlohmann::json repodata_record = m_package_info;
 
-        if (index.find("size") == index.end() || index["size"] == 0)
+        // To take correction of packages metadata (e.g. made using repodata patches) into account,
+        // we insert the index into the repodata record to only add new fields from the index
+        // while keeping the existing fields from the repodata record.
+        repodata_record.insert(index.cbegin(), index.cend());
```

Comment:
Did you find a confirmation that `insert` will ignore values that already exist? I couldnt find a confirmation in [the documentation](https://json.nlohmann.me/api/basic_json/insert/).
Very minor concern, as I suspect this is correct and just not well documented.

Reply:
See: https://github.com/nlohmann/json/blob/85df7ed59350563c36e39ba999212229ba4bfb1c/include/nlohmann/ordered_map.hpp#L329-L340

Reply:
Unfortunately implementations are not intent so it's best when it's the documentation clarifying the pre-conditions and post-conditions clearly.

No need to add the link, I'm mentionning that in doubt I didnt search properly but I believe the issue is that the libraryu should clarify it in the doc so that solid assumptions can be built over it. No need to comment in the code about this.

Reply:
I was not able to find it in the documentation as well.